parsl/tests/run.sh

```diff
@@ -0,0 +1,52 @@
+export PARSL_TESTING="true"
+
+set -e
+
+flake8 parsl/
+bash parsl/tests/lint-inits.sh
+
+# This uses all of the configurations and tests as the base from which to
+# run mypy checks - these are likely to capture most of the code used in
+# parsl
+MYPYPATH=$(pwd)/mypy-stubs mypy parsl/tests/configs/
+MYPYPATH=$(pwd)/mypy-stubs mypy parsl/tests/test*/
+MYPYPATH=$(pwd)/mypy-stubs mypy parsl/tests/sites/
+
+# do this before any testing, but not in-between tests
+rm -f .coverage
+
+pytest parsl -k "not cleannet" --config parsl/tests/configs/htex_local.py --cov=parsl --cov-append --cov-report= --random-order
+pytest parsl -k "not cleannet" --config parsl/tests/configs/local_threads.py --cov=parsl --cov-append --cov-report= --random-order
+
+# some of the site/ tests require more dependencies. These are installed here as needed,
+# so that the above tests happen with only the basic requirements installed.
+
+# workqueue
+./parsl/executors/workqueue/install-workqueue.sh
```

Comment:
this and MPI install at line 30 don't need to happen on each run if a user is running this script in a persistent environment - maybe this can be parameterised on the commandline to make it happen or not? If I'm going to use this on the commandline in my dev environment I want this to run the tests, not mess round with what is installed, most of the time.